sdk/core/corehttp/tests/perf_tests/_async_iterator_random_stream.py

```diff
@@ -0,0 +1,71 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+from devtools_testutils.perfstress_tests._random_stream import get_random_bytes, _DEFAULT_LENGTH
+
+
+# The AsyncIteratorRandomStream is used specifically for the AsyncHttpXTransport upload stream scenario, since the
+# async httpx transport requires the request body stream to be type AsyncIterator (i.e. have an __aiter__ and __anext__ method rather than __iter__).
+# Specific check in httpx here: https://github.com/encode/httpx/blob/7df47ce4d93a06f2c3310cd692b4c2336d7663ba/httpx/_content.py#L116.
+
+class AsyncIteratorRandomStream():
```

Comment:
Probably not super important, but worth explicitly stating that this is an `AsyncIterator`.

```suggestion
class AsyncIteratorRandomStream(AsyncIterator[bytes]):
```